### PR TITLE
kernelpkg updates

### PR DESCRIPTION
**Pull request by amendlik**
### What does this PR do?
1) Fixes an issue where kernelpkg.list_installed on YUM produced an unordered list
2) Adds a new function: kernelpkg.upgrade_available

### What issues does this PR fix or reference?
None

### Tests written?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: salt/modules/kernelpkg_linux_apt.py [hunk 1]
```diff
@@ -179,6 +179,20 @@ def upgrade(reboot=False, at_time=None):
     return ret
 
 
+def upgrade_available():
+    '''
+    Detect if a new kernel version is available in the repostories.
+    Returns True if a new kernal is avaliable, False otherwise.
```
Comment (pgporada) [inline salt/modules/kernelpkg_linux_apt.py]: `kernal` should be `kernel`

File: salt/modules/kernelpkg_linux_yum.py [hunk 2]
```diff
@@ -167,8 +166,36 @@ def upgrade(reboot=False, at_time=None):
     return ret
 
 
+def upgrade_available():
+    '''
+    Detect if a new kernel version is available in the repostories.
+    Returns True if a new kernal is avaliable, False otherwise.
```
Comment (pgporada) [inline salt/modules/kernelpkg_linux_yum.py]: `kernal` should be `kernel`

**Review by cachedout — CHANGES_REQUESTED** [review]
Waiting on spelling fixes pointed out by @pgporada

File: salt/modules/kernelpkg_linux_yum.py (hunk 2 above)
Comment (pgporada) [inline salt/modules/kernelpkg_linux_yum.py]: `repostories` should be `repositories`

File: salt/modules/kernelpkg_linux_apt.py (hunk 1 above)
Comment (pgporada) [inline salt/modules/kernelpkg_linux_apt.py]: `repostories` should be `repositories`

**Pull request merged**